src/components/Stories/Stories.tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+
+import {block} from '../utils/cn';
+import {Modal, ModalCloseReason} from '../Modal';
+import {Button} from '../Button';
+import {ButtonClose} from '../Dialog/ButtonClose/ButtonClose';
+import {Link} from '../Link';
+import {MediaRenderer} from './components';
+import {Story} from './types';
+import i18n from './i18n';
+
+import './Stories.scss';
+
+const b = block('stories');
+
+export interface StoriesProps {
+    open: boolean;
+    stories: Story[];
+    onClose?: (
+        event: MouseEvent | KeyboardEvent | React.MouseEvent<HTMLElement, MouseEvent>,
+        reason: ModalCloseReason | 'closeButtonClick',
+    ) => void;
+    startStoryIndex?: number;
+    onPreviousClick?: (storyIndex: number) => void;
+    onNextClick?: (storyIndex: number) => void;
+}
+
+export function Stories({
+    open,
+    onClose,
+    stories,
+    onPreviousClick,
+    onNextClick,
+    startStoryIndex,
+}: StoriesProps) {
+    const startIndex = React.useMemo(() => {
+        if (
+            typeof startStoryIndex === 'undefined' ||
+            startStoryIndex < 0 ||
+            startStoryIndex >= stories.length
+        ) {
+            return 0;
+        }
+
+        return startStoryIndex;
+    }, [startStoryIndex, stories]);
+    const [currentStoryIndex, setCurrentStoryIndex] = React.useState(startIndex);
+
+    const currentStory = stories[currentStoryIndex];
+    const isFirstStory = currentStoryIndex === 0;
+    const isLastStory = currentStoryIndex === stories.length - 1;
+    const hasNextStory = !isLastStory;
+    const hasPreviousStory = !isFirstStory;
+
+    const handleClose = React.useCallback<NonNullable<StoriesProps['onClose']>>(
+        (event, reason) => {
+            onClose?.(event, reason);
+        },
+        [onClose],
+    );
+
+    const handleButtonClose = React.useCallback<
+        (event: MouseEvent | KeyboardEvent | React.MouseEvent<HTMLElement, MouseEvent>) => void
+    >(
+        (event) => {
+            handleClose(event, 'closeButtonClick');
+        },
+        [handleClose],
+    );
+
+    const handleGotoPrevious = React.useCallback(() => {
+        if (currentStoryIndex > 0) {
+            const newIndex = currentStoryIndex - 1;
+            setCurrentStoryIndex(newIndex);
+            onPreviousClick?.(newIndex);
+        }
+    }, [currentStoryIndex, onPreviousClick]);
+
+    const handleGotoNext = React.useCallback(() => {
+        if (currentStoryIndex < stories.length - 1) {
+            const newIndex = currentStoryIndex + 1;
+            setCurrentStoryIndex(newIndex);
+            onNextClick?.(newIndex);
+        }
+    }, [currentStoryIndex, stories, onNextClick]);
+
+    return (
+        <Modal open={open} onClose={handleClose} className={b('modal')}>
+            <div className={b('wrap')}>
+                <div className={b()}>
```

Comment:
What is the reason for 2 div's? Inner div is just make rounded corners and copy some width/height styles from parent div

Reply:
This is necessary for the correct background color, component centering and size setting.